tests/test_db_runtime_security.py

```
@@ -144,914 +173,937 @@ def _connect_side_effect(*args, **kwargs):
+
+
+# ========================================================================
+# Connection Error / Pragma Application Tests
+# ========================================================================
+
+
+def test_connection_creation_error_cleanup(db_path):
+    """Test that connection creation errors don't leak connections."""
+    # Try to create a manager with invalid pragma that will fail
+    # If __init__ raises, nothing to clean up
+    with pytest.raises(ValueError):
```

Ruff (0.15.12)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)